hw05-bytecode/src/main/java/skislitsyn/utils/Ioc.java

```diff
@@ -0,0 +1,73 @@
+package skislitsyn.utils;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.HashSet;
+
+import skislitsyn.logging.Log;
+
+public class Ioc {
+    private Ioc() {
+    }
+
+    public static Object createMyClass(Object obj) {
+	InvocationHandler handler = new MyInvocationHandler(obj);
+	return Proxy.newProxyInstance(Ioc.class.getClassLoader(), obj.getClass().getInterfaces(), handler);
+    }
+
+    static class MyInvocationHandler implements InvocationHandler {
+	private final Object myClass;
+	private final HashMap<Class<?>, HashSet<Method>> loggableInterfaces = new HashMap<>();
```

Comment:
Здесь можно просто:
```suggestion
	private final HashMap<Method> loggableInterfaces = new HashMap<>();
```

`MyInvocationHandler ` всегда связан только с одним классом.
При новом вызове `Ioc.createMyClass` будет создаваться новый `MyInvocationHandler `, который всегда будет связан только с  переданным `myClass`.

Кстати почему `myClass`? Там будет класс? Или объект? Не обманывает ли имя переменной?

Reply:
действительно, этот момент не уловил - заменил мап на набор.
название переменной скорректировал, забавно, что оно перекочевало сюда из примера Сергея :)